### PR TITLE
Transmit CI environment variable in metrics

### PR DESCRIPTION
**Pull request by nbrahms**
We want to understand the environments in which Semgrep is run, in order
to:
- improve Semgrep value by building for one-off search vs. continuous
  scanning use cases
- correlate errors to environments

Our PRIVACY.md actually included a CI field in its field descriptions,
but this was previously unimplemented.

PR checklist:
- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by nbrahms** [comment]
cc @minusworld for security review

**Comment by minusworld** [comment]
> cc @minusworld for security review

Eh, I think it's fine 👍

File: semgrep/semgrep/metric_manager.py [hunk 1]
```diff
@@ -191,6 +192,7 @@ def as_dict(self) -> Dict[str, Any]:
                 "projectHash": self._project_hash,
                 "configNamesHash": self._configs_hash,
                 "rulesHash": self._rules_hash,
+                "ci": os.environ.get("CI"),
```
Comment (emjin) [inline semgrep/semgrep/metric_manager.py:195]: Would anyone ever set CI to anything other than true or false? Would it be a good idea to validate this?

**Review by emjin — APPROVED** [review]

**Comment by brendongo** [comment]
Can we link to documentation to github actions, gitlab actions etc that say CI is set? Can't seem to easily find documentation on when/what this env var is set to

**Review by brendongo — APPROVED** [review]

**Comment by nbrahms** [comment]
> Can we link to documentation to github actions, gitlab actions etc that say CI is set? Can't seem to easily find documentation on when/what this env var is set to

- GitHub: CI=true (https://docs.github.com/en/actions/learn-github-actions/environment-variables)
- GitLab: CI=true (https://docs.gitlab.com/ee/ci/variables/)
- CircleCI: CI=true (https://circleci.com/docs/2.0/env-vars/)
- Travis CI: CI=true (https://docs.travis-ci.com/user/environment-variables/)

File: semgrep/semgrep/metric_manager.py (hunk 1 above)
Comment (nbrahms) [reply semgrep/semgrep/metric_manager.py:195]: I think we should just broadly accept whatever the CI provider does, so Semgrep is more robust

**Pull request merged**